pkg/obfuscator/obfuscator.go

```diff
@@ -2,9 +2,10 @@ package obfuscator
 
 // Obfuscator is the interface which all obfuscators should implement
 type Obfuscator interface {
-	ReplacementReporter
 	// FileName takes a filename as input and return the obfuscated name
 	FileName(string) string
 	// Contents takes string as input and return the obfuscated string
 	Contents(string) string
+	// ReportingResult returns a map of words and their replacements
+	ReportingResult() map[string]string
```

Comment:
interesting choice, why not just refactor the `ReplacementReporter` to have an atomic report+get?

Reply:
or have an atomic replacement reporter and kick the simple one out of the project

Reply:
It's not that easy. The reporter would need to have a generator which generates a replacement when there is no existing replacement. Because the new replacement should be generated atomically as well. So I move the entire logic into the `ipGenerator`. Also IMO it doesn't make sense for an obfuscator to expose methods like `ReportReplacement`.

Reply:
what if you pass a function for the generation? eg like https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent(K,java.util.function.Function)

Reply:
That's possible. But I don't like that then the `GetReplacement` will return a replacement in the generating reporter and in the `SimpleReporter` it won't. If we want to refactor this we should do it separately.

Reply:
okay cool, I will send you a PR then with the refactoring later today. let's get this merged :)